src/test/java/io/fabric8/maven/docker/BuildMojoTest.java

```diff
@@ -241,6 +244,75 @@ void buildUsingBuildxWithMultipleAuth() throws IOException, MojoExecutionExcepti
         thenAuthContainsRegistry("custom-registry.org");
     }
 
+    void buildUsingBuildxWithTag(boolean skipTag) throws IOException, MojoExecutionException {
+        givenBuildXService();
+
+        List<String> tags = new ArrayList<>();
+        tags.add("tag-" + System.currentTimeMillis());
+        tags.add("tag-" + System.currentTimeMillis());
+
+        givenMavenProject(buildMojo);
+        ImageConfiguration imageConfiguration = singleImageConfiguration(builder -> {
+            builder.buildx(getBuildXPlatforms(TWO_BUILDX_PLATFORMS).build());
+            builder.tags(tags);
+            builder.skipTag(skipTag);
+        });
+        givenResolvedImages(buildMojo, Collections.singletonList(imageConfiguration));
+        givenPackaging("jar");
+
+        whenMojoExecutes();
+
+        List<String> fullTags = skipTag ? Collections.emptyList() : tags.stream()
+                .map(tag -> new ImageName(imageConfiguration.getName(), tag).getFullName())
+                .collect(Collectors.toList());
+        thenBuildxRun(null, null, true, null, fullTags);
+    }
+
+    @Test
+    void buildUsingBuildxWithTag() throws IOException, MojoExecutionException {
+        buildUsingBuildxWithTag(false);
+    }
+
+    @Test
+    void buildUsingBuildxWithSkipTag() throws IOException, MojoExecutionException {
+        buildUsingBuildxWithTag(true);
+    }
+
+    void buildWithTag(boolean skipTag) throws IOException, MojoExecutionException {
+        givenMavenProject(buildMojo);
+
+        List<String> tags = new ArrayList<>();
+        tags.add("tag-" + System.currentTimeMillis());
+        tags.add("tag-" + System.currentTimeMillis());
+
+        ImageConfiguration imageConfiguration = singleImageConfiguration(builder -> {
+            builder.skipTag(skipTag);
+            builder.tags(tags);
+        });
+        givenResolvedImages(buildMojo, Collections.singletonList(imageConfiguration));
+        givenPackaging("jar");
+        givenSkipPom(true);
+
+        whenMojoExecutes();
+
+        thenBuildRun();
+
+        if (!skipTag) {
+            Mockito.verify(buildService, Mockito.times(1))
+                    .tagImage(Mockito.any(ImageConfiguration.class));
+        }
+    }
+
+    @Test
+    void buildWithTag() throws IOException, MojoExecutionException {
+        buildWithTag(false);
+    }
+
+    @Test
+    void buildWithSkipTag() throws IOException, MojoExecutionException {
+        buildWithTag(true);
+    }
```

Comment:
Is it possible to somehow use `@Nested` annotation to group normal build and buildX tests? Initially, I was a bit confused by the test names.  Can we use JUnit5 parameterized tests here to reduce duplication?

Reply:
@rohanKanojia Done.